[jishnu28] iP

jishnu28 opened the pull request:
### Duke (by Jishnu)

Duke is a chatbot/task list manager with the following features:
- Add tasks to list
- Supports tasks of 3 types (Todos, Deadlines, Events)
- Delete tasks from list
- Mark tasks as completed or incomplete


To try out Duke simply follow these steps:
1. Download Duke.jar from [here](https://github.com/jishnu28/ip/releases)
2. Open a new terminal
3. Navigate to the folder containing Duke.jar
4. Run the command java -jar {filename}.jar e.g., java -jar Duke.jar (i.e., run the command in the same folder as the jar file)


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
    public class Main {
        public static void main(String[] args) {
            Application.launch(MainApp.class, args);
        }
    }
```

**Progress of features:** 🙌
- [x]  Week 1 - 3 features
- [ ]  Week 4 - 6 features

> "I like Duke" - This could be you!

Commit:
Add Gradle support

jishnu28 committed:
Completed Level-1

jishnu28 committed:
Added Level-2 features

jishnu28 committed:
Added Level-3 features

jishnu28 committed:
Added Level-4 features

jishnu28 committed:
Added A-TextUiTesting features

jishnu28 committed:
Added Level-5 features

jishnu28 committed:
Added Level-6 features

jishnu28 committed:
Fix coding standard

jishnu28 committed:
Added header comments

jishnu28 committed:
Added Level-7 features

jishnu28 committed:
Added Level-8 features

jishnu28 committed:
Fixed Merge conflicts

jishnu28 committed:
Added A-MoreOOP features

jishnu28 committed:
Added A-Packages features

jishnu28 committed:
Merging add-gradle-support to master to get gradle wrapper files
Merge branch 'add-gradle-support'

jishnu28 committed:
Added A-Gradle features

jishnu28 committed:
Added A-JUnit features

jishnu28 committed:
Added A-Jar features

jishnu28 committed:
Add Level-9 feature to branch

RyanQiu1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+     * @param taskList TaskList object to which tasks are added to or deleted from
+     * @param fileManager Storage object that deals with loading tasks from file and saving tasks in file
+     */
+    public void parse(String userInput, TaskList taskList, Storage fileManager) {
```
I like the way the parser is thought out here, but have perhaps using cases could make the code cleaner.

RyanQiu1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,104 @@
+ */
+public class Parser {
+    /**
+     * Makes sense of the user command and calls corresponding methods
```
Perhaps this description could be more descriptive?

RyanQiu1 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,43 @@
+/**
+ * Class that defines the Event type of tasks
+ */
+public class Event extends Task {
```
Good naming of variables and methods👍

RyanQiu1 commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,24 @@
+     * Testing the encode method of the Deadline class
+     */
+    @Test
+    public void encodeTest(){
```
Missing a space here after the method name.

RyanQiu1 commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,23 @@
+     * Testing the encode method of the Todo class
+     */
+    @Test
+    public void encodeTest(){
```
Missing a space here after the method name.

RyanQiu1 commented on src/test/java/duke/TodoTest.java:
```diff
@@ -0,0 +1,23 @@
+    }
+
+    @Test
+    public void anotherDummyTest(){
```
Missing a space here after the method name.

RyanQiu1 reviewed (commented):
Readable and easily understandable code, however some coding standard slips, almost there. Just a few nits to fix.👍

jengoc415 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,68 @@
+    protected String title;
+    /** Type of Task */
+    protected char taskType; // can be T, D or E
+
```
Good naming of variables

jengoc415 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,38 @@
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello I'm\n" + logo);
```
Good use of line break to make code more readable

jengoc415 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,83 @@
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
```
Good consistent usage of 'this', neat

jengoc415 reviewed (commented):
Coding standards are adhered to. Code is very readable. Great job!

jishnu28 committed:
Add Level-10 feature to branch

jishnu28 committed:
Add A-Assertions features

jishnu28 committed:
Merge pull request #2 from jishnu28/branch-A-Assertions

Add A-Assertions features

jishnu28 committed:
Add A-CodeQuality features

jishnu28 committed:
Merge pull request #3 from jishnu28/branch-A-CodeQuality

Add A-CodeQuality features

jishnu28 committed:
Edit assert statements to improve debug message

jishnu28 committed:
Edit header comments for class descriptions to better describe the class

jishnu28 committed:
Merge pull request #4 from jishnu28/branch-A-Assertions

Edit assert statements to improve debug message

jishnu28 committed:
Merge pull request #5 from jishnu28/branch-A-CodeQuality

Edit header comments for class descriptions to better describe the class

jishnu28 committed:
Commands to create new tasks and mark them as complete or incomplete require full taskType and instruction to be specified in command

This makes the commands long-winded and difficult to type

Let's provide new shorter aliases for the keywords for the following commands:
* d -> deadline
* t -> todo
* e -> event
* m -> mark
* um -> unmark
* del -> delete

This makes the commands easier to type

jishnu28 committed:
Merge pull request #6 from jishnu28/branch-C-FriendlierSyntax

Add C-FriendlierSyntax features

jishnu28 committed:
Add A-UserGuide specifications

jishnu28 committed:
Modify build.gradle to set Launcher class as entrypoint

jishnu28 committed:
Add Exception management for delete command

jishnu28 committed:
Merge pull request #7 from jishnu28/branch-A-MoreErrorHandling

Add Exception management for delete command

jishnu28 committed:
Add A-Personality features and adjust GUI

jishnu28 committed:
Merge pull request #8 from jishnu28/branch-A-Personality

Add A-Personality features and adjust GUI

jishnu28 committed:
Edit userguide to include more information
